internal/ethapi/api.go

```diff
@@ -1004,6 +1005,16 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(hash common.Hash) (map[
 		"logs":              receipt.Logs,
 		"logsBloom":         receipt.Bloom,
 	}
+
+	// Assign receipt status or post state.
+	if receipt.PostState != nil {
+		fields["root"] = hexutil.Bytes(receipt.PostState)
+	} else {
+		fields["statusCode"] = receiptStatusSuccessful
+		if receipt.Failed {
+			fields["statusCode"] = receiptStatusFailed
+		}
+	}
```

Comment:
Receipt already has `statusEncoding()`. Can't we expose that one and use it instead?

Also, the EIP describes the status as a single byte, either 0 or 1. In the RPC bytes are hex encoded so this should be something like: `fields["statusCode"] = hexutil.Bytes(receipt.StatusEncoding())`. The EIP also doesn't mention that the field name is `statusCode`, that should probably be amended to the EIP.

Reply:
@bas-vk, I had purposed a new [EIP](https://github.com/ethereum/EIPs/pull/699) about adding a new field named `statusCode` to rpc returned. And as you mentioned, all rpc fields should be hex encoded, i think it is correct, and i will make change to satisfy this requirement.